ansible_collections/f5networks/f5_modules/plugins/modules/bigip_device_info.py

```diff
@@ -8919,6 +8919,57 @@ def read_collection_from_device(self):
         return result
 
 
+class DOParameters(BaseParameters):
+    api_map = {
+    }
+
+    returnables = [
+
+    ]
+
+
+class DOFactManager(BaseManager):
+    def __init__(self, *args, **kwargs):
+        self.client = kwargs.get('client', None)
+        self.module = kwargs.get('module', None)
+        super(DOFactManager, self).__init__(**kwargs)
+        self.want = self.module.params
+
+    def exec_module(self):
+        facts = self._exec_module()
+        result = dict(do_config=facts)
+        return result
+
+    def _exec_module(self):
+        results = []
+        facts = self.read_facts()
+        return facts
+
+    def read_facts(self):
+        results = []
+        collection = self.read_collection_from_device()
+        return collection
+
+    def read_collection_from_device(self):
+        uri = "https://{0}:{1}/mgmt/shared/declarative-onboarding/config".format(
+            self.client.provider['server'],
+            self.client.provider['server_port'],
+        )
+        resp = self.client.api.get(uri)
+        try:
+            response = resp.json()
+        except ValueError as ex:
+            raise F5ModuleError(str(ex))
+
+        if resp.status not in [200, 201] or 'code' in response and response['code'] not in [200, 201]:
+            raise F5ModuleError(resp.content)
+
+        if 'parsed' not in response[0]:
+            return []
+        result = response[0]
+        return result
```

Comment:
This should be a GET to /declare

Reply:
There's no endpoint called declarative-onboarding/declare
There's config, info, inspect, and task - 

https://github.com/F5Networks/f5-declarative-onboarding/blob/master/src/schema/latest/openapi.yaml

Reply:
There is /declare endpoint but it works with POST only. 

What I mean was to get the declaration you send GET to this endpoint:
`/mgmt/shared/declarative-onboarding/`

```{
    "id": "85062824-d37a-41f4-b99f-150d30430733",
    "selfLink": "https://localhost/mgmt/shared/declarative-onboarding/task/85062824-d37a-41f4-b99f-150d30430733",
    "result": {
        "class": "Result",
        "code": 200,
        "status": "OK",
        "message": "success"
    },
    "declaration": {
        "schemaVersion": "1.2.0",
        "class": "Device",
        "async": true,
        "Common": {
            "class": "Tenant",
            "hostname": "bigip.example.com",
            "myProvisioning": {
                "class": "Provision",
                "ltm": "nominal"
            }
        }
    }
}```

We are only interested what is under declaration

Reply:
Ok, this is odd... When I opened the base URL on my setup, I got this - 
```
{"code":422,"message":"request failed with null exception","originalRequestBody":"{\"id\":\"342f24a7-f19d-458c-a4c1-76b79a811b28\",\"selfLink\":\"https://localhost/mgmt/shared/declarative-onboarding/task/342f24a7-f19d-458c-a4c1-76b79a811b28\",\"code\":422,\"status\":\"ERROR\",\"message\":\"invalid config - rolled back\",\"errors\":[\"01070734:3: Configuration error: invalid Management Route, the dest/netmask pair 0.0.0.0/0.0.0.0 already exists for /Common/default\",\"01070734:3: Configuration error: invalid Management Route, the dest/netmask pair 0.0.0.0/0.0.0.0 already exists for /Common/default\"],\"result\":{\"class\":\"Result\",\"code\":422,\"status\":\"ERROR\",\"message\":\"invalid config - rolled back\",\"errors\":[\"01070734:3: Configuration error: invalid Management Route, the dest/netmask pair 0.0.0.0/0.0.0.0 already exists for /Common/default\",\"01070734:3: Configuration error: invalid Management Route, the dest/netmask pair 0.0.0.0/0.0.0.0 already exists for /Common/default\"]},\"declaration\":{\"webhook\":\"https://example.com/myHook\",\"label\":\"my BIG-IP declaration for declarative onboarding\",\"Common\":{\"hostname\":\"bigip.example.com\",\"external-self\":{\"allowService\":\"none\",\"vlan\":\"external\",\"trafficGroup\":\"traffic-group-local-only\",\"class\":\"SelfIp\",\"address\":\"10.20.0.100/24\"},\"managementRoute\":{\"gw\":\"1.2.3.4\",\"class\":\"ManagementRoute\",\"network\":\"default\",\"mtu\":1500},\"admin\":{\"userType\":\"regular\",\"shell\":\"bash\",\"class\":\"User\",\"keys\":[]},\"myDns\":{\"nameServers\":[\"8.8.8.8\",\"2001:4860:4860::8844\"],\"search\":[\"f5.com\"],\"class\":\"DNS\"},\"anotherUser\":{\"userType\":\"regular\",\"shell\":\"none\",\"class\":\"User\",\"partitionAccess\":{\"all-partitions\":{\"role\":\"guest\"}},\"keys\":[]},\"myNtp\":{\"timezone\":\"UTC\",\"class\":\"NTP\",\"servers\":[\"0.pool.ntp.org\",\"1.pool.ntp.org\",\"2.pool.ntp.org\"]},\"myProvisioning\":{\"class\":\"Provision\",\"ltm\":\"nominal\",\"gtm\":\"minimum\"},\"default\":{\"gw\":\"10.10.0.1\",\"class
```

So it does show the declaration section, but it also throws a 422. I thought this meant I'm not supposed to do a GET on this.

Reply:
Is this fine or do you still believe we should be getting data from the /declare endpoint?